### ISSUE TITLE
Unit tests are broken on master

### ISSUE THREAD

**Issue by JustinBeckwith** [googleapis/nodejs-spanner#829]
Just cloning the repo and running the tests, looks like the unit tests are somehow broken.  PRs are starting to pile up - we should prioritize ASAP.  

@olavloite @skuruppu

**Comment by olavloite** [comment]
@skuruppu @JustinBeckwith I'll have a look right away.

**Comment by olavloite** [comment]
@JustinBeckwith @skuruppu @bcoe 
The problem appears to be related to Node version 12.16.0. I had 12.15.0 installed locally and could run all tests without problems. I've now installed 12.16.0 and get the error that is also showing up in the CI:

```
  1) Database
       "before each" hook for "should promisify all the things":
     TypeError: Cannot assign to read only property 'errorMonitor' of function 'class Database extends service_object_1.GrpcServiceObject {
    constructor(instance, name, poolOptions) {
    ...<omitted>...
}'
      at setProperty (node_modules/extend/index.js:46:24)
      at extend (node_modules/extend/index.js:108:7)
      at Context.<anonymous> (test/database.ts:218:5)
      at processImmediate (internal/timers.js:456:21)
```

**Comment by ljharb** [comment]
Reposting here from https://github.com/googleapis/nodejs-spanner/pull/830#issuecomment-585925678:
> While this is a fine change, the issue isn't `extend`; it's that you're doing this: https://github.com/googleapis/nodejs-spanner/blob/master/test/database.ts#L35 which mutates, and should cause `Object.assign` to throw as well in strict mode.
> 
> "promisify all" patterns that mutate should be avoided.

**Comment by bcoe** [comment]
this was ultimately fixed in `Node@12.16.1`, just a fun note: https://nodejs.org/en/blog/release/v12.16.1/
